### PR TITLE
Tkt 35. Add markers on brackets.

### PR DESCRIPTION
**Pull request by Pochatkin**
Need add marker 'mismatch'

### CODE REVIEW

**Comment by dbarashev** [comment]
Pls use descriptive names in the pull requests.

File: lib/ace/editor.js [hunk 1]
```diff
@@ -535,8 +535,26 @@ var Editor = function(renderer, session) {
             } else if (session.$mode.getMatching) {
                 var range = session.$mode.getMatching(self.session);
             }
-            if (range)
+            if (range) {
                 session.$bracketHighlight = session.addMarker(range, "ace_bracket", "text");
+            } else {
+                pos = session.findSurroundingBrackets(self.getCursorPosition());
+                if (!pos.mismatch) {
+                    if (pos) {
```
Comment (dbarashev) [inline lib/ace/editor.js]: you have already checked that pos is not null at the previous line

File: lib/ace/editor.js [hunk 2]
```diff
@@ -535,8 +535,26 @@ var Editor = function(renderer, session) {
...
+                        if (pos.left && pos.right) {
```
Comment (dbarashev) [inline lib/ace/editor.js]: besides, you have a guarantee that mismatch == false => left and right are defined

File: lib/ace/editor.js [hunk 3]
```diff
@@ -535,8 +535,26 @@ var Editor = function(renderer, session) {
...
+                            var rangeLeft = new Range(pos.left.row, pos.left.column, pos.left.row, pos.left.column + 1);
+                            var rangeRight = new Range(pos.right.row, pos.right.column, pos.right.row, pos.right.column + 1);
+                        }
+                    }
+
+                    if (rangeLeft && rangeRight) {
```
Comment (dbarashev) [inline lib/ace/editor.js]: these vars are out of current scope

File: lib/ace/editor.js (hunk 1 above)
Comment (Pochatkin) [inline lib/ace/editor.js]: Yeah, i khow about it. Already made refactoring.

File: lib/ace/editor.js (hunk 3 above)
Comment (Pochatkin) [inline lib/ace/editor.js]: It's too.

File: lib/ace/editor.js [hunk 4]
```diff
@@ -535,8 +533,36 @@ var Editor = function(renderer, session) {
...
+                        var rangeLeft = new Range(pos.left.row, pos.left.column, pos.left.row, pos.left.column + 1);
+                        var rangeRight = new Range(pos.right.row, pos.right.column, pos.right.row, pos.right.column + 1);
+                        //TODO: Need add marker mismatch. I don't know how.
+                    }
+                    if (pos.left) {
+                        var rangeLeft = new Range(pos.left.row, pos.left.column, pos.left.row, pos.left.column + 1);
+                        //TODO: Need add marker mismatch. I don't know how.
+                    }
+                    if (pos.right) {
+                        var rangeRight = new Range(pos.left.row, pos.left.column, pos.left.row, pos.left.column + 1);
```
Comment (dbarashev) [inline lib/ace/editor.js]: hello copy && paste. s/left/right

**Comment by dbarashev** [comment]
LGTM [1 comment]

**Pull request merged**